plugin/storage/grpc/factory.go

```diff
@@ -128,7 +128,7 @@ func (f *Factory) newRemoteStorage(
 		bearertoken.NewUnaryClientInterceptor(),
 	}
 	streamInterceptors := []grpc.StreamClientInterceptor{
-		tenancy.NewClientStreamInterceptor(tenancyMgr),
+		bearertoken.NewStreamClientInterceptor(),
```

Comment:
good catch. Can you move `tenancyMgr` variable creation down, just before it is used?